src/govuk/components/notification-banner/template.test.js

```diff
@@ -0,0 +1,276 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-env jest */
+
+const axe = require('../../../../lib/axe-helper')
+
+const { render, getExamples } = require('../../../../lib/jest-helpers')
+
+const examples = getExamples('notification-banner')
+
+describe('Notification-banner', () => {
+  describe('default example', () => {
+    it('passes accessibility tests', async () => {
+      const $ = render('notification-banner', examples.default)
+
+      const results = await axe($.html())
+      expect(results).toHaveNoViolations()
+    })
+
+    it('aria-labelledby attribute matches the title id', () => {
+      const $ = render('notification-banner', examples.default)
+      const ariaAttr = $('.govuk-notification-banner').attr('aria-labelledby')
+
+      expect(ariaAttr).toEqual('govuk-notification-banner-title')
+    })
+
+    it('has role=region attribute', () => {
+      const $ = render('notification-banner', examples.default)
+      const $component = $('.govuk-notification-banner')
+
+      expect($component.attr('role')).toEqual('region')
+    })
+
+    it('does not have tabindex set', () => {
+      const $ = render('notification-banner', examples.default)
+
+      const $component = $('.govuk-notification-banner')
+      expect($component.attr('tabindex')).toBeFalsy()
+    })
+
+    it('does not have data-module attribute to focus component on page load', () => {
+      const $ = render('notification-banner', examples.default)
+      const $component = $('.govuk-notification-banner')
+
+      expect($component.attr('data-module')).not.toEqual('govuk-initial-focus')
+    })
+
+    it('renders header container', () => {
+      const $ = render('notification-banner', examples.default)
+      const $header = $('.govuk-notification-banner__header')
+
+      expect($header.length).toBeTruthy()
+    })
+
+    it('renders default heading level', () => {
+      const $ = render('notification-banner', examples.default)
+      const $title = $('.govuk-notification-banner__title')
+
+      expect($title.get(0).tagName).toEqual('h2')
+    })
+
+    it('renders default title text', () => {
+      const $ = render('notification-banner', examples.default)
+      const $title = $('.govuk-notification-banner__title')
+
+      expect($title.html().trim()).toEqual('Important')
+    })
+
+    it('title id matches the aria-labelledby attribute on the component', () => {
```

Comment:
At the minute, you could just change the ID, fix the hardcoded ID in the test, and end up with an aria-labelledby that points to an invalid ID. 

Would it make sense to have one test that ties the two together? I think that'd also mean we could cover this relationship with one test rather than two.

```js
    it('aria-labelledby attribute matches the title id', () => {
      const $ = render('notification-banner', examples.default)
      const ariaAttr = $('.govuk-notification-banner').attr('aria-labelledby')
      const titleId = $('.govuk-notification-banner__title').attr('id')

      expect(ariaAttr).toEqual(titleId)
    })
```

We've done something sort of similar in the checkboxes template tests:

https://github.com/alphagov/govuk-frontend/blob/abe1672b68fe08a55fb3562a684b408a5d452bb0/src/govuk/components/checkboxes/template.test.js#L336-L347

Reply:
Done!